### PR TITLE
Restrict making of containers to non-infinite ranges

### PR DESCRIPTION
**Pull request by nordlow**
Model-checking that forbids construction of containers from infinite generators.

### CODE REVIEW

File: std/container/util.d [hunk 1]
```diff
@@ -103,22 +103,32 @@ unittest
...
     }
 }
 
+/// forbid construction from infinite range
+unittest
+{
+    import std.container.array : Array;
+    import std.range : repeat;
+    import std.range.primitives : isInfinite;
+    static assert(!__traits(compiles, { auto arr = make!Array(repeat(5)); }));
```
Comment (MetaLang) [inline std/container/util.d]: Would be nice to have the positive case tested as well. Something like:

`static assert(__traits(compiles, { auto arr = make!Array(only(5)); }));`

**Comment by MetaLang** [comment]
Urgh, this is going to take some getting used to.

**Comment by nordlow** [comment]
@MetaLang What do you mean, getting used to?

**Comment by MetaLang** [comment]
I'm referring to the new Github review system. This is my first time using it and it's going to take some getting used to ;-)

**Comment by nordlow** [comment]
Aha.

**Comment by nordlow** [comment]
Two has approved...so why no merge?

**Comment by jmdavis** [comment]
Auto-merge toggled on

**Comment by nordlow** [comment]
Thanks.

**Comment by nordlow** [comment]
@MetaLang `array()` seems protected against infinite ranges. Any more functions that need to be qualified?

**Comment by MetaLang** [comment]
Not sure, I'm guessing we'll just have to cross those bridges as we come to them. The only other functions I can think of at the moment are writeln et al. They'll happily take an infinite range and print out its elements forever.

**Comment by wilzbach** [comment]
> All checks have passed
> 5 successful checks

Seems like auto-tester has still now some troubles with automatically merging (it worked on a other PR), let's have an eye on this. I just deprecated one build which will hopefully trigger the merge soon.

**Comment by schveiguy** [comment]
It's possible another PR merged. Github doesn't have great sync with auto tester, it sometimes keeps old status.

**Pull request merged**